### DISCUSSION TITLE
line gaps if time series interval is irregular

### DISCUSSION THREAD

**Discussion by miloaissatu** [predict-idlab/plotly-resampler#299] [Q&A] [answered]
python3.11, plotly_resampler 0.9.2
```
import numpy as np
import pandas as pd

from plotly_resampler import register_plotly_resampler

register_plotly_resampler(mode='widget')
pd.options.plotting.backend = 'plotly'


x = pd.DataFrame(np.random.normal(0, 0.01, 4000))
x.index = pd.Timestamp('2023-01-01') + (pd.Timedelta('1s') * np.random.rand(4000) * 20)
x = x.sort_index()
x[0] = x[0].cumsum()

x.plot(line_shape='hv')
```
![image](https://github.com/predict-idlab/plotly-resampler/assets/8089596/38b63ffe-af6d-4a23-ba0b-62e41f8c1019)

when plotting a time series with irregular intervals there's line gaps when zooming is. is this expected behaviour? are there any configs to have it connect the datapoints?

**Comment by jonasvdd** [comment, marked as answer]
Hi @miloaissatu,

Yes there is! 
You can use the `NoGapHandler` ([docs](https://predict-idlab.github.io/plotly-resampler/v0.9.2/api/aggregation/gap_handlers/)), which can be passed to the `default_gap_handler` property of `AbstractFigureAggregator` its constructor ([docs](https://predict-idlab.github.io/plotly-resampler/v0.9.2/api/figure_resampler/figure_resampler_interface/#figure_resampler.figure_resampler_interface.AbstractFigureAggregator.__init__)). 

```python
import numpy as np; import pandas as pd

from plotly_resampler import register_plotly_resampler
# NOTE: I added this import to the original example
from plotly_resampler.aggregation import NoGapHandler


# And added the `default_gap_handler` argument to the `register_plotly_resampler` call
register_plotly_resampler(mode='widget', default_gap_handler=NoGapHandler())
pd.options.plotting.backend = 'plotly'


x = pd.DataFrame(np.random.normal(0, 0.01, 4000))
x.index = pd.Timestamp('2023-01-01') + (pd.Timedelta('1s') * np.random.rand(4000) * 20)
x = x.sort_index()
x[0] = x[0].cumsum()

x.plot(line_shape='hv')
```


Hope this resolves your issue,
Jonas

**Reply by miloaissatu** [reply]
awesome. tyvm @jonasvdd

**Reply by boxingshark** [reply]
That helped me a lot! Thanks!
